### PR TITLE
bump argument parser to 1.2.x

### PR DESCRIPTION
**Pull request by tomerd**

### CODE REVIEW

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131

@swift-ci test

**Review by ahoppen — APPROVED** [review]

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131

@swift-ci test

File: Package.swift [hunk 1]
```diff
@@ -244,7 +244,7 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
     .package(name: "IndexStoreDB", url: "https://github.com/apple/indexstore-db.git", .branch("main")),
     .package(name: "SwiftPM", url: "https://github.com/apple/swift-package-manager.git", .branch("main")),
     .package(url: "https://github.com/apple/swift-tools-support-core.git", .branch("main")),
-    .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.1.4")),
+    .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.2.0")),
```
Comment (tomerd) [inline Package.swift:247]:
```suggestion
    .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMinor(from: "1.2.2")),
```

**Comment by tomerd** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131

@swift-ci test

**Review by ahoppen — APPROVED** [review]

**Comment by shahmishal** [comment]
https://github.com/apple/swift/pull/63521
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131

@swift-ci test Windows

**Pull request merged**